packages/ipfs-cpinner-provider/src/IpfsPinnerProvider.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Content, CID, DID, IpfsPinner, Key, MetadataManager, IpfsClient
+} from './types'
+
+export default class IpfsPinnerProvider {
+  constructor (
+    private ipfsClient: IpfsClient,
+    private metadataManager: MetadataManager,
+    private ipfsPinner: IpfsPinner
+  ) {}
+
+  async put (did: DID, key: Key, content: Content): Promise<CID> {
+    const cid = await this.ipfsClient.put(content)
+    await this.ipfsPinner.pin(cid)
+    await this.metadataManager.save(did, key, cid)
+    return cid
+  }
+
+  async get (did: DID, key: Key): Promise<Content[]> {
+    const cids = await this.metadataManager.find(did, key)
+
+    const promises = []
+    for (let i = 0; i < cids.length; i++) {
+      const cid = cids[i]
+      promises.push(this.ipfsClient.get(cid))
+    }
+
+    return Promise.all(promises).then(arrays => Array.prototype.concat.apply([], arrays))
```

Comment:
Can we use `Array.map`? Is more descriptive. I would write

```suggestion
    return Promise.all(cids.map(cid => this.ipfsClient.get(cid))).then(arrays => Array.prototype.concat.apply([], arrays))
```